course_catalog/migrations/0044_uniq_by_platform_id.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 2.1.11 on 2019-10-07 19:34
+
+from django.db import migrations, models
+
+from course_catalog.constants import OFFERED_BY_MAPPINGS
+
+
+def populate_platforms(apps, schema_editor):
+    """
+    Set all avatars to None. Since we don't show avatars yet
+    this avoids the work of populating avatar_small and avatar_medium
+    """
+    Course = apps.get_model("course_catalog", "Course")
+    LearningResourceRun = apps.get_model("course_catalog", "LearningResourceRun")
+
+    for course in Course.objects.iterator():
+        course.platform = OFFERED_BY_MAPPINGS[course.offered_by]
+        course.save()
+
+    for run in LearningResourceRun.objects.iterator():
+        run.platform = OFFERED_BY_MAPPINGS[course.offered_by]
+        run.save()
```

Comment:
Assuming all runs have a populated `offered_by` field, I think this should be:
```python
    for run in LearningResourceRun.objects.iterator():
        run.platform = OFFERED_BY_MAPPINGS[run.offered_by]
        run.save()
```

if not, then:

```python
    Course = apps.get_model("course_catalog", "Course")
    Bootcamp = apps.get_model("course_catalog", "Bootcamp")
    Program = apps.get_model("course_catalog", "Program")
    LearningResourceRun = apps.get_model("course_catalog", "LearningResourceRun")

    for run in LearningResourceRun.objects.filter(content_type__model="course").iterator():
        course = Course.objects.get(id=run.object_id)
        course.platform = OFFERED_BY_MAPPINGS[course.offered_by]
        course.save()
        run.platform = OFFERED_BY_MAPPINGS[course.offered_by]
        run.save()

    for run in LearningResourceRun.objects.filter(content_type__model="bootcamp").iterator():
        bootcamp = Bootcamp.objects.get(id=run.object_id)
        bootcamp.platform = OFFERED_BY_MAPPINGS[bootcamp.offered_by]
        bootcamp.save()
        run.platform = OFFERED_BY_MAPPINGS[bootcamp.offered_by]
        run.save()

    for run in LearningResourceRun.objects.filter(content_type__model="program").iterator():
        program = Program.objects.get(id=run.object_id)
        run.platform = OFFERED_BY_MAPPINGS[program.offered_by]
        run.save()
```

I was getting some dupe runs with the code as is, unless I deleted everything and imported from scratch.

Reply:
We want the first one, I also made it a bit safer by switching to `.get()` with the current value as the default in case there's some weird cases where `offered_by` isn't set/

Reply:
I checked, there's no null or empty values in prod, so we shouldn't see an issue, but probably better safe than sorry...